MAUI/Autocomplete/Searching-Filtering.md

```diff
@@ -171,6 +171,276 @@ autocomplete.TextSearchMode = AutocompleteTextSearchMode.Contains;
+
+* `source` - The owner of the search behavior, which holds information about `ItemsSource` and so on.
+* `searchInfo` - Contains information about the filtered items based on the text entered in `Autocomplete` control. You can select item from the drop-down list using this list. 
+
+The following code demonstrates initially how to select an item from the filtered list of drop-down in the `Autocomplete` control. When entering the country's name, the capital cities will be selected.
+{% tabs %}
+
+{% highlight C# %}
+
+public class CapitalCitySearchingBehavior : Syncfusion.Maui.Inputs.IAutocompleteSearchBehavior
+{
+    public int GetHighlightIndex(SfAutocomplete source, AutocompleteSearchInfo searchInfo)
+    {
+
+        var filteredCapitals = from CityInfo cityInfo in searchInfo.FilteredItems
+                               where cityInfo.IsCapital
+                               select searchInfo.FilteredItems.IndexOf(cityInfo);
+        if (filteredCapitals.Count() > 0)
+            return filteredCapitals.FirstOrDefault();
+
+        return 0;
+    }
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+**Step 3:** Applying custom searching to `Autocomplete` control by using the `SearchBehavior` property. 
+
+{% tabs %}
+{% highlight XAML %}
+ <ContentPage.BindingContext>
+        <local:CityViewModel></local:CityViewModel>
+    </ContentPage.BindingContext>
+<editors:SfAutocomplete 
+    DisplayMemberPath="CityName"
+    ItemsSource="{Binding Cities}">
+            <editors:SfAutocomplete.FilterBehavior>
+                <local:CityFilteringBehavior/>
+            </editors:SfAutocomplete.FilterBehavior>
+<editors:SfAutocomplete.SearchBehavior>
+                <local:CapitalCitySearchingBehavior></local:CapitalCitySearchingBehavior>
+            </editors:SfAutocomplete.SearchBehavior>
+</editors:SfAutoComplete>
+
+{% endhighlight %}
+{% endtabs %}
+
+The following gif demonstrates how to select the capital city from the drop-down based on the country name entered in the `Autocomplete` control.
+
+![.NET MAUI AutoComplete filter the items based on custom filtering and searching logic.](Images/Searching_Filtering/Custom_filtering_searching.gif)
+
+## Load asynchronous items
+
+Load the data dynamically at runtime based on typed input. This dynamic loading can be done while performing custom filtering using the `CustomFilter` property.
+
+The `GetMatchingItemsAsync` method of `IAutocompleteFilterBehavior` helps you to perform filtering operations on different threads without blocking the current thread by using `await Task.Run()`.
+
+**Step 1:** Create a class that derives from the `IAutocompleteFilterBehavior` interface and add your custom filter logic in `GetMatchingItemsAsync` method to load the run time items based on typed input.
```

Comment:
**Step 1:** Create a class from the `IAutocompleteFilterBehavior` interface and add your custom filter logic to the `GetMatchingItemsAsync` method to load the runtime items based on typed input.